iceoryx_binding_c/source/c_client.cpp

```diff
@@ -71,16 +71,18 @@ iox_client_t iox_client_init(iox_client_storage_t* self,
     if (options != nullptr)
     {
         clientOptions.responseQueueCapacity = options->responseQueueCapacity;
-        clientOptions.nodeName = iox::NodeName_t(TruncateToCapacity, options->nodeName);
+        clientOptions.nodeName = iox::NodeName_t(
+            TruncateToCapacity, options->nodeName, strnlen(options->nodeName, iox::NodeName_t::capacity()));
```

Comment:
@elfenpiff @FerdinandSpitzschnueffler From my point of view it's ok to use `strnlen` on a user-provided string as `iceoryx_binding_c` is not [quality level 1+](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#quality-levels).